algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/treatment/HasHadTreatmentResponseFollowingSomeTreatmentOrCategoryOfTypes.kt

```diff
@@ -48,19 +49,33 @@ class HasHadClinicalBenefitFollowingSomeTreatmentOrCategoryOfTypes(
             }
         }
 
-        val treatmentsWithResponse = targetTreatmentsToResponseMap.filterKeys { it in BENEFIT_RESPONSE_SET }.values.flatten()
-        val benefitMessage = " objective benefit from treatment"
+        val evaluateClinicalBenefit = treatmentResponses == TreatmentResponse.BENEFIT_RESPONSES
+
+        val treatmentsWithResponse = targetTreatmentsToResponseMap.filterKeys { it in treatmentResponses }.values.flatten()
+        val otherResponses = targetTreatmentsToResponseMap.filterKeys { it !in treatmentResponses }.keys.filterNotNull()
```

Comment:
You could use `partition` here so only one pass is needed:
```suggestion
        val (treatmentsWithResponse, otherResponses) = targetTreatmentsToResponseMap.entries.partition { it.key in treatmentResponses }
            .let { (withResponse, otherResponse) -> withResponse.flatMap { it.value } to otherResponse.mapNotNull { it.key } }
```